gen3-integration-tests/gen3_ci/scripts/select_ci_environment.py

```diff
@@ -34,12 +34,10 @@ def select_ci_environment(namespaces):
         if status == "Completed":
             return job.get_artifact_content(build_num, "namespace.txt")
         else:
-            logger.error("Build timed out. Consider increasing max_duration")
             job.terminate_build(build_num)
-            return None
+            raise Exception("Build timed out. Consider increasing max_duration")
```

Comment:
We need a return value here since that is used to condition whether to run the pipeline steps to unlock CI environment. If we want to do this then we should also make changes to the GH Actions workflow logic. Maybe we should keep these in separate PRs - the tests and CI improvements

Reply:
ok, it's not a great flow to have to check for None returned values upstream (see [this](https://cdis.slack.com/archives/C06CV20211P/p1704829237124249?thread_ts=1704827642.458079&cid=C06CV20211P)), i think Exceptions make more sense, but we can see how to get it working with the workflow

Reply:
Agreed, I was having serious trouble with the Github actions condition checks, found that this worked ... but now the current logic seemed to have stopped working as well. We will have to rethink the logic anyway.